service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -2041,49 +2041,55 @@ public void saveAdminCommentToOrder(AdminCommentDto adminCommentDto, String uuid
      * @param orderId      {@link Long}.
      * @param uuid         {@link String}.
      *
-     * @author Yuriy Bahlay.
+     * @author Yuriy Bahlay, Sikhovskiy Rostyslav.
      */
     @Override
-    public void updateEcoNumberForOrder(List<EcoNumberDto> ecoNumberDto, Long orderId, String uuid) {
-        User currentUser = userRepository.findUserByUuid(uuid)
+    public void updateEcoNumberForOrder(EcoNumberDto ecoNumberDto, Long orderId, String uuid) {
+        final User currentUser = userRepository.findUserByUuid(uuid)
             .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
         Order order = orderRepository.findById(orderId).orElseThrow(
             () -> new OrderNotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + orderId));
-        if (ecoNumberDto != null) {
-            StringBuilder collectedValue = new StringBuilder();
-            for (int i = 0; i < ecoNumberDto.size(); i++) {
-                EcoNumberDto ecoNumber = ecoNumberDto.get(i);
-                String oldNumber = orderRepository.findEcoNumberFromShop(ecoNumber.getOldEcoNumber(), orderId);
-                if (oldNumber != null) {
-                    orderRepository.setOrderAdditionalNumber(ecoNumber.getNewEcoNumber(), oldNumber, orderId);
-                    collectedValue.append(
-                        collectInfoAboutEcoNumberEventHistory(i, oldNumber, ecoNumber.getNewEcoNumber()));
-                }
-            }
-            if (collectedValue.length() > 1) {
-                eventService.save(collectedValue.toString(),
-                    currentUser.getRecipientName() + "  " + currentUser.getRecipientSurname(), order);
-            }
-        }
+        Set<String> oldEcoNumbers = Set.copyOf(order.getAdditionalOrders());
```

Comment:
```suggestion
           @Override
    public void updateEcoNumberForOrder(EcoNumberDto ecoNumberDto, Long orderId, String uuid) {
        final User currentUser = userRepository.findUserByUuid(uuid)
            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
        Order order = orderRepository.findById(orderId).orElseThrow(
            () -> new OrderNotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST + orderId));
        Set<String> oldEcoNumbers = Set.copyOf(order.getAdditionalOrders());
        Set<String> newEcoNumbers = ecoNumberDto.getEcoNumber();

        Collection<String> removed = CollectionUtils.subtract(oldEcoNumbers, newEcoNumbers);
        Collection<String> added = CollectionUtils.subtract(newEcoNumbers, oldEcoNumbers);
        
        StringBuilder eventString = new StringBuilder();
        eventString.append(getHistorySummaryString(added, OrderHistory.ADD_NEW_ECO_NUMBER))
                .append(getHistorySummaryString(removed, OrderHistory.DELETED_ECO_NUMBER));
        
        orderRepository.save(order);
        eventService.save(eventString.toString().trim(),
            currentUser.getRecipientName() + "  " + currentUser.getRecipientSurname(), order);
    }
    
    private String getHistorySummaryString(Collection<String> items, String sectionTitle) {
        return String.format("%s: %s", sectionTitle, String.join("; ", items));
    }
```


please include  dependency
```suggestion
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-collections4</artifactId>
            <version>4.4</version>
        </dependency>
```
it's pretty useful

Reply:
Please use this code as a reference and implement in a similar way

Reply:
fixed